File: backend/src/main/java/com/hotel/reservation/service/GuestService.java
package com.hotel.reservation.service;

import com.hotel.reservation.dto.GuestDTO;
import com.hotel.reservation.dto.GuestRequest;
import java.util.List;

/**
 * Service interface for Guest operations.
 */
public interface GuestService {
    GuestDTO createGuest(GuestRequest request);
    GuestDTO updateGuest(Long id, GuestRequest request);
    void deleteGuest(Long id);
    GuestDTO getGuestById(Long id);
    List<GuestDTO> getAllGuests();
    List<GuestDTO> getGuestsByUser(Long userId);
}

File: frontend/src/services/guestService.ts
import apiClient from './apiService';
import { Guest, GuestRequest } from '../utils/types';

export const guestService = {
  async getAllGuests(): Promise<Guest[]> {
    const response = await apiClient.get<Guest[]>('/guests');
    return response.data;
  },

  async getGuestById(id: number): Promise<Guest> {
    const response = await apiClient.get<Guest>(`/guests/${id}`);
    return response.data;
  },

  async createGuest(guest: GuestRequest): Promise<Guest> {
    const response = await apiClient.post<Guest>('/guests', guest);
    return response.data;
  },

  async updateGuest(id: number, guest: Partial<GuestRequest>): Promise<Guest> {
    const response = await apiClient.put<Guest>(`/guests/${id}`, guest);
    return response.data;
  },

  async deleteGuest(id: number): Promise<void> {
    await apiClient.delete(`/guests/${id}`);
  },

  async getGuestByUserId(userId: number): Promise<Guest> {
    const response = await apiClient.get<Guest[]>(`/guests/user/${userId}`);
    // Return first guest or throw error if no guest found
    if (response.data && response.data.length > 0) {
      return response.data[0];
    }
    throw new Error('No guest found for this user');
  }
};
